### PR TITLE
Fix trafficgen metadata

### LINKED ISSUE #1079: pbench-trafficgen data is currently not indexable
The pbench server only indexes data from pbench tar balls which have a properly formatted `metadata.log` with all the requisite data.

The only benchmark that is not providing that data is `pbench-trafficgen`.

### PR DESCRIPTION
**Pull request by portante**
Fixes #1079.

### CODE REVIEW

**Comment by portante** [comment]
> Is adding `pbench-collect-sysinfo` what generates the metadata log? Or was there something else that changed that fixed this?

It is the call to `pbench-metadata-log` with `beg` and `end` that does it.  I updated the script to match the behavior of the other benchmarks, but since trafficgen has never collected any system information by default, I made the default for `pbench-collect-sysinfo` be `none`, but add the parameter `--sysinfo` so that folks who actually want to collect system information like the other benchmarks can.

File: agent/bench-scripts/gold/pbench-trafficgen/test-21.txt [hunk 1]
```diff
@@ -228,6 +228,10 @@ options common in most pbench benchmark scripts
 
 --pre-sample-cmd=str
   run this command before executing each test sample
+
+--sysinfo=str,
+  comma seperated values of sysinfo to be collected, default="none"
```
Comment (ndokos) [inline agent/bench-scripts/gold/pbench-trafficgen/test-21.txt]: separated

File: agent/bench-scripts/pbench-trafficgen [hunk 2]
```diff
@@ -346,6 +347,10 @@ if [ $? -ne 0 ]; then
 	printf -- "\n"
 	printf -- "--pre-sample-cmd=str\n"
 	printf -- "  run this command before executing each test sample\n"
+	printf -- "\n"
+	printf -- "--sysinfo=str,\n"
+	printf -- "  comma seperated values of sysinfo to be collected, default=\"none\"\n"
```
Comment (ndokos) [inline agent/bench-scripts/pbench-trafficgen]: separated

File: agent/bench-scripts/pbench-trafficgen [hunk 3]
```diff
@@ -1149,9 +1178,9 @@ else
 	done
 fi
 if [ $rc -eq 0 ]; then
-	echo "$script_path/postprocess/generate-benchmark-summary $benchmark_name $orig_cmd $benchmark_run_dir" >"$benchmark_run_dir/generate-benchmark-summary.cmd"
+	echo "$script_path/postprocess/generate-benchmark-summary \"$benchmark_name\" \"$orig_cmd\" \"$benchmark_run_dir\"" >"$benchmark_run_dir/generate-benchmark-summary.cmd"
+	chmod +x "$benchmark_run_dir/generate-benchmark-summary.cmd"
```
Comment (ndokos) [inline agent/bench-scripts/pbench-trafficgen]: Why is "$benchmark_run_dir/generate-benchmark-summary.cmd" generated at all? For documentation? Should it be used on line 1183?

File: agent/bench-scripts/gold/pbench-trafficgen/test-26.txt [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+--tool-group=str
+  Start/stop/post-process tools using this group
+
+--pre-sample-cmd=str
+  run this command before executing each test sample
+
+--sysinfo=str,
+  comma seperated values of sysinfo to be collected, default="none"
+    available: default,none,all,a,b,c,d,e
```
Comment (ndokos) [inline agent/bench-scripts/gold/pbench-trafficgen/test-26.txt]: I presume this uses a mock version of pbench-collect-sysinfo?

**Comment by ndokos** [comment]
BTW, the last sentence of the commit message needs some massaging.

File: agent/bench-scripts/gold/pbench-trafficgen/test-26.txt (hunk 4 above)
Comment (portante) [reply agent/bench-scripts/gold/pbench-trafficgen/test-26.txt]: Yes, I believe so.

File: agent/bench-scripts/pbench-trafficgen (hunk 3 above)
Comment (portante) [reply agent/bench-scripts/pbench-trafficgen]: Seems to be an artifact of how all these benchmark scripts were created.  Added a commit to fix them all.

File: agent/bench-scripts/pbench-trafficgen (hunk 2 above)
Comment (portante) [reply agent/bench-scripts/pbench-trafficgen]: Fixed.

File: agent/bench-scripts/gold/pbench-trafficgen/test-21.txt (hunk 1 above)
Comment (portante) [reply agent/bench-scripts/gold/pbench-trafficgen/test-21.txt]: Fixed.

**Pull request merged**